File: Day 1 - Easy (Level 1)/MostProfitable/Solution/MostProfitable.js
var test_case1 = {

    cost : [100,120,150,1000],
    price : [110,110,200,2000],
    sales : [20,100,50,3],
    items : ["Video Card","256M Mem","CPU/Mobo combo","Complete machine"],
    expected : "Complete machine"

},

test_case2 = {

    cost : [100],
    price : [100],
    sales : [134],
    items : ["Service, at cost"],
    expected : " "

}

test_case3 = {

    cost : [38,24],
    price : [37,23],
    sales : [1000,643],
    items : ["Letter","Postcard"],
    expected : " "

}

test_case4 = {

    cost : [10,10],
    price : [11,12],
    sales : [2,1],
    items : ["A","B"],
    expected : "A"

}

console.log("\n Case 1, Expected Result: "+test_case1.expected+", Output: "+bestItem(test_case1.cost,test_case1.price,test_case1.sales,test_case1.items));
console.log("\n Case 2, Expected Result: "+test_case2.expected+", Output: "+bestItem(test_case2.cost,test_case2.price,test_case2.sales,test_case2.items));
console.log("\n Case 3, Expected Result: "+test_case3.expected+", Output: "+bestItem(test_case3.cost,test_case3.price,test_case3.sales,test_case3.items));
console.log("\n Case 4, Expected Result: "+test_case4.expected+", Output: "+bestItem(test_case4.cost,test_case4.price,test_case4.sales,test_case4.items));

function bestItem(cost,price,sales,items){
    //Variable declaration and initialization
	var n = cost.length;
    var MaxProfit = 0;
    var MostProfitable = 0;

    //Itearate through all items
    for (var i = 0; i < n; i++) {
        var profit = (price[i] - cost[i]) * sales[i]; //Calculating profit for each item
        if (profit > MaxProfit) {
        MaxProfit = profit; 	//Storing most profitable iteme and its index.
        MostProfitable = i;
        }
    }
    if (MaxProfit == 0)
        return "";
    else
        return items[MostProfitable];
}

File: Day 1 - Easy (Level 1)/MostProfitable/MostProfitable.cs
// BEGIN CUT HERE

// END CUT HERE
using System;
using System.Text;
using System.Text.RegularExpressions;
using System.Collections;

public class MostProfitable {
    public string bestItem(int[] costs, int[] prices, int[] sales, string[] items) {
        string res = "";
        return res;
    }

// BEGIN CUT HERE
    public static void Main(String[] args) {
        try  {
            eq(0,(new MostProfitable()).bestItem(new int[] {100,120,150,1000}, new int[] {110,110,200,2000}, new int[] {20,100,50,3}, new string[] {"Video Card","256M Mem","CPU/Mobo combo","Complete machine"}),"Complete machine");
            eq(1,(new MostProfitable()).bestItem(new int[] {100}, new int[] {100}, new int[] {134}, new string[] {"Service, at cost"}),"");
            eq(2,(new MostProfitable()).bestItem(new int[] {38,24}, new int[] {37,23}, new int[] {1000,643}, new string[] {"Letter","Postcard"}),"");
            eq(3,(new MostProfitable()).bestItem(new int[] {10,10}, new int[] {11,12}, new int[] {2,1}, new string[] {"A","B"}),"A");
        } 
        catch( Exception exx)  {
            System.Console.WriteLine(exx);
            System.Console.WriteLine( exx.StackTrace);
        }
    }
    private static void eq( int n, object have, object need) {
        if( eq( have, need ) ) {
            Console.WriteLine( "Case "+n+" passed." );
        } else {
            Console.Write( "Case "+n+" failed: expected " );
            print( need );
            Console.Write( ", received " );
            print( have );
            Console.WriteLine();
        }
    }
    private static void eq( int n, Array have, Array need) {
        if( have == null || have.Length != need.Length ) {
            Console.WriteLine("Case "+n+" failed: returned "+have.Length+" elements; expected "+need.Length+" elements.");
            print( have );
            print( need );
            return;
        }
        for( int i= 0; i < have.Length; i++ ) {
            if( ! eq( have.GetValue(i), need.GetValue(i) ) ) {
                Console.WriteLine( "Case "+n+" failed. Expected and returned array differ in position "+i );
                print( have );
                print( need );
                return;
            }
        }
        Console.WriteLine("Case "+n+" passed.");
    }
    private static bool eq( object a, object b ) {
        if ( a is double && b is double ) {
            return Math.Abs((double)a-(double)b) < 1E-9;
        } else {
            return a!=null && b!=null && a.Equals(b);
        }
    }
    private static void print( object a ) {
        if ( a is string ) {
            Console.Write("\"{0}\"", a);
        } else if ( a is long ) {
            Console.Write("{0}L", a);
        } else {
            Console.Write(a);
        }
    }
    private static void print( Array a ) {
        if ( a == null) {
            Console.WriteLine("<NULL>");
        }
        Console.Write('{');
        for ( int i= 0; i < a.Length; i++ ) {
            print( a.GetValue(i) );
            if( i != a.Length-1 ) {
                Console.Write(", ");
            }
        }
        Console.WriteLine( '}' );
    }
// END CUT HERE
}
